cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -204,13 +217,13 @@ impl Cmd {
 #[async_trait::async_trait]
 impl NetworkRunnable for Cmd {
     type Error = Error;
-    type Result = TxnResult<String>;
+    type Result = Either<String, TxnResult<String>>;
```

Comment:
I don't know a lot about the `either` crate, but what is the benefit of using it over creating our own enum?

Reply:
We can create a separate enum I originally though it'll only be used in one place and there's much reason of doing so